libraries/azurerm_subnet.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'azurerm_resource'
+
+class AzurermSubnet < AzurermSingularResource
+  name 'azurerm_subnet'
+  desc 'Verifies settings for an Azure Virtual Network Subnet'
+  example <<-EXAMPLE
+    describe azurerm_subnet(resource_group: 'example',vnet: 'virtual-network-name' name: 'subnet-name') do
+      it { should exist }
+      its('name') { should eq 'subnet-name' }
+    end
+  EXAMPLE
+
+  ATTRS = %i(
+    id
+    name
+    type
+    properties
+  ).freeze
+
+  attr_reader(*ATTRS)
+
+  def initialize(resource_group: nil, vnet: nil, name: nil)
+    resp = client.subnet(resource_group, vnet, name)
+    return if resp.nil? || resp.key?('error')
+
+    ATTRS.each do |field|
+      instance_variable_set("@#{field}", resp[field.to_s])
+    end
```

Comment:
We have a new helper for here as well:
```
  assign_fields(ATTRS, resp)
```
This will handle assigning the fields and defining new methods. Source here: https://github.com/inspec/inspec-azure/blob/master/libraries/azurerm_resource.rb#L50-L56